source/MRViewer/MRSurfaceContoursWidget.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "MRViewer.h"
+#include "MRViewerEventsListener.h"
+#include "MRMesh/MRMeshFwd.h"
+#include "MRSurfacePointPicker.h"
+#include <unordered_map>
+#include "MRMesh/MRObjectMeshHolder.h"
+
+
+namespace MR
+{
+
+class SurfaceContoursWidget : public MultiListener<
+    MouseDownListener,
+    MouseMoveListener>
+{
+public:
+
+    using PickerPointCallBack = std::function<void( std::shared_ptr<MR::ObjectMeshHolder> )>;
+    using SurfaceContour = std::vector<std::shared_ptr<SurfacePointWidget>>;
+    using SurfaceContours = std::unordered_map <std::shared_ptr<MR::ObjectMeshHolder>, SurfaceContour>;
+
+    MRVIEWER_API bool onMouseDown_( Viewer::MouseButton button, int modifier ) override;
+    MRVIEWER_API bool onMouseMove_( int mouse_x, int mouse_y ) override;
+
+    // enable or disable widget
+    MRVIEWER_API void enable( bool isEnaled );
+
+    // create a widget and connect it. 
+    MRVIEWER_API void create( PickerPointCallBack onPointAdd, PickerPointCallBack onPointMove, PickerPointCallBack onPointMoveFinish, PickerPointCallBack onPointRemove );
+
+    // clear temp internal variables.
+    MRVIEWER_API void clear();
+
+    // reset widget, clear internal variables and detach from signals.
+    MRVIEWER_API void reset();
+
+    // return contour for specific object.
+    [[nodiscard]] const SurfaceContour& getSurfaceContour( const std::shared_ptr<MR::ObjectMeshHolder> obj )
```

Comment:
if const why not `&`
also should fuction be const too? (also should we check that obj is present in the map?)

Reply:
sure